File: Day-7/Question_1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Day_7
{
    class Question_1
    {
        public int Partition(int[] arr, int left, int right)
        {
            int pivot;
            pivot = arr[left];
            while (true)
            {
                while (arr[left] < pivot)
                {
                    left++;
                }
                while (arr[right] > pivot)
                {
                    right--;
                }
                if (left < right)
                {
                    int temp = arr[right];
                    arr[right] = arr[left];
                    arr[left] = temp;
                }
                else
                {
                    return right;
                }
            }
        }
        public void quickSort(int[] arr, int left, int right)
        {
            int pivot;
            if (left < right)
            {
                pivot = Partition(arr, left, right);
                if (pivot > 1)
                {
                    quickSort(arr, left, pivot - 1);
                }
                if (pivot + 1 < right)
                {
                    quickSort(arr, pivot + 1, right);
                }
            }
        }
        public int CalculateDistance(int[] point)
        {
            return point[0] * point[0] + point[1] * point[1];
        }
        public int[][] KClosest(int[][] points, int K)
        {
            // int[][] result = new int[K][];
            // int[] distances = new int[points.Length];
            // int index = 0;
            // foreach(int[] point in points){
            //     int distance = CalculateDistance(point);
            //     distances[index++]=distance;
            // }
            // // quickSort(distances,0, distances.Length-1);
            // int index_2 = 0;
            // foreach(int[] point in points){
            //     if(CalculateDistance(point)<=distances[K-1])
            //         result[index_2++] = point;
            // }
            return points
            .OrderBy(p => Math.Pow(Math.Abs(p[0]), 2) + Math.Pow(Math.Abs(p[1]), 2))
            .Take(K)
            .ToArray();
        }
    }
}

File: Day-6/Question_1.c
using System;
using System.Collections.Generic;
using System.Text;

namespace Day_6
{
	class Question_1
	{
		public int[] SortArrayByParityII(int[] A) {
			int max_digit = 0;
			for (int i = 0; i < A.Length; i++)
			{
				if (max_digit < A[i]) max_digit = A[i];
			}
			int[] accumulation_array = new int[max_digit + 1];
			for (int i = 0; i < accumulation_array.Length; i++)
			{
				accumulation_array[i] = 0;
			}
			for (int i = 0; i < A.Length; i++)
			{
				accumulation_array[A[i]] += 1;
			}
			int[] result_array = new int[A.Length];
			int index = 0;
			for (int i = 0; i < accumulation_array.Length; i++)
			{
				if (accumulation_array[i] == 0) continue;
				for (int j = 0; j < accumulation_array[i]; j++)
				{
					result_array[index] = i;
					index += 1;
				}
			}
			for (int i = 0; i < result_array.Length; i++)
			{
				int current_number = result_array[i];
				if (current_number % 2 == 0 && i % 2 == 0) continue;
				else if (current_number % 2 != 0 && i % 2 != 0) continue;
				else
				{
					for (int j = i + 1; j < result_array.Length; j++)
					{
						if (i % 2 == 0) {
							if (result_array[j] % 2 == 0)
							{
								int temp = result_array[i];
								result_array[i] = result_array[j];
								result_array[j] = temp;
								break;
							}
						}
						else {
							if (result_array[j] % 2 != 0)
							{
								int temp = result_array[i];
								result_array[i] = result_array[j];
								result_array[j] = temp;
								break;
							}
						}
					}
				}
			}
			return result_array;

		}
	}
}
